gateway/src/service.rs

```diff
@@ -772,14 +783,49 @@ pub mod tests {
             }
         );
         assert_eq!(
-            svc.iter_user_projects_detailed(neo.clone())
+            svc.iter_user_projects_detailed(&neo, 0, u32::MAX)
                 .await
                 .unwrap()
                 .map(|item| item.0)
                 .collect::<Vec<_>>(),
             vec![matrix.clone()]
         );
 
+        let mut all_projects: Vec<ProjectName> = (1..60)
+            .map(|p| ProjectName(format!("matrix-{p}")))
+            .collect();
+        for p in &all_projects {
+            svc.create_project(p.clone(), neo.clone(), false, 0)
+                .await
+                .unwrap();
+        }
+        all_projects.insert(0, matrix.clone());
+
+        assert_eq!(
+            svc.iter_user_projects_detailed(&neo, 0, u32::MAX)
+                .await
+                .unwrap()
+                .map(|item| item.0)
+                .collect::<Vec<_>>(),
+            all_projects
+        );
+        assert_eq!(
+            svc.iter_user_projects_detailed(&neo, 0, 20)
+                .await
+                .unwrap()
+                .map(|item| item.0)
+                .collect::<Vec<_>>(),
+            all_projects[..20]
+        );
+        assert_eq!(
+            svc.iter_user_projects_detailed(&neo, 20, 20)
```

Comment:
Same as before, what would happen if we pass an out-of-bounds offset?

Reply:
I'll keep this open as well because would be good to include a case where we query for an offset greater than the rows count.